google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/HmacKey.java

```diff
@@ -0,0 +1,287 @@
+package com.google.cloud.storage;
+
+import com.google.api.client.util.DateTime;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+public class HmacKey implements Serializable {
+
+  private static final long serialVersionUID = -1809610424373783062L;
+  private final String secretKey;
+  private final HmacKeyMetadata metadata;
+
+  private HmacKey(Builder builder) {
+    this.secretKey = builder.secretKey;
+    this.metadata = builder.metadata;
+  }
+
+  public static Builder newBuilder(String secretKey) {
+    return new Builder(secretKey);
+  }
+
+  public static class Builder {
+    private String secretKey;
```

Comment:
Not super familiar with documenting Java code, but can we add a note saying the `secret` is only available on creation?

Reply:
yeah--i'm planning to add the documentation once the code is otherwise approved